docs/css/main.css

```diff
@@ -56,28 +56,28 @@ mark {
     margin: 0 auto;
     min-width: 0;
     max-width: 1000px;
+    overflow-x: auto;
```

Comment:
Documenting the layout file changes for review + future release changelog noting (should be simplified though I think =P)

1st of 3 important changes in the layout file:

This might be somewhat controversial, but is unfortunately necessary with a sticky header. At least, I haven't found another solution.


The issue (without this) occurs on pages that overflow (e.g. "formatting contents" page or "Using html javascript and css"). The [nearest scrolling ancestor](https://developer.mozilla.org/en-US/docs/Web/CSS/position) which is the page / root element expands to accomodate the overflown content, which in turn does 2 things:
- `window.innerWidth/Height` are erroneous (although there is an easy alternative for this `document.documentElement.clientWidth/Height`)
- More important is that the way sticky positioning is defined, the sticky positioning goes haywire (with or without our header height detection)

Gif demo (open image in new tab to view in order): there are 3 settings here.

1. The first shows proper function of sticky positioning when the viewport zooms out to the entire page.
2. The second showcases how the page loads on a mobile device normally, observe how the sticky positioning completely goes haywire as the page is "zoomed" (although it dosen't quite appear like that to the reader).
3. The third is same as the second, except I temporarily toggled off the header hiding (the `translateY`) to show the issue is really unrelated to header hiding / anything we are doing.

![stickyissues](https://user-images.githubusercontent.com/3306138/173222485-42af641c-1d61-4228-8d90-a5222ea0844e.gif)

You can try removing the `overflow-x` when testing this PR to see the issue better, its a little hard to present on a gif.

The fix: I've set `overflow-x` on the content wrapper which significantly changes ux on mobile devices (but just pages that overflow).

Pros:
- As a nice side effect, this change "fixes" https://github.com/MarkBind/markbind/issues/1741.
- Overflown content is still usable by swiping left / right.

Cons:
- On desktop devices, although _very very_ unlikely (since there is enough space for overflown content on mobile to fully "expand" - including our embeds), if there is overflow, I would still like to point out the horizontal scroll bar is only visible at the bottom of the content wrapper. (which can be confusing)
   - add to that that site development should usually be done on a desktop device (so authors will likely limit content sizes to a reasonable degree (e.g. images))
- You'll no longer be able to "zoom out" on mobile (though arguably this could also be a plus for mobile ux)

Not sure if there's a better solution =\, but imo its a good enough compromise.